internal/restorable/image.go

```diff
@@ -257,7 +257,7 @@ func (i *Image) makeStale(rect image.Rectangle) {
 	}
 
 	// Remove duplicated regions to avoid unnecessary reading pixels from GPU.
-	n := removeDuplicatedRegions(i.staleRegions)
+	n := removeDuplicatedRegions(i.staleRegions, origNum)
```

Comment:
Would this work instead of modifying `removeDuplicatedRegions`?

```go
n := removeDuplicatedRegions(i.staleRegions[origNum:])
i.staleRegions = i.staleRegions[:len(origNum)+n]
```

Reply:
I realized my suggestion is different from the fix. I'll take a look at your PR tomorrow...

Reply:
Yeah - basically, the idea is that in the entire list, there is no pair of rectangles for which each is contained in the other.

My change turns this into an invariant, i.e. assumes that before adding more, this property is already the case, so we only need to check for these duplicates if EITHER the inner OR the outer rectangle is among the new ones.